### PR TITLE
Announce the correct line when on the end of a link at the end of a list item in a contenteditable

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
None.

### Summary of the issue:
If a web page contains a contenteditable, which contains an inline element at the end of a block-level element E.g. a link at the end of a paragraph, or a span at the end of an li (list item) element,
And the caret is placed at the very end of the line, reading the line with NVDA will announce the next line, not the current. Similarly, if downArrowing to the next line and then back up again, the next line is announced, not the current.

This can be reproduced in Google Docs documents, by simply creating a list with one or more items, and placing a link as the end of the top list item.
E.g.
* I like [NVDA](https://www.nvaccess.org/)
* I like cats

or this html fragment:
```
data:text/html,<div contenteditable="true"><ul><li>I like <a href="http://www.nvaccess.org/">NVDA</a<</li><li>I like cats</li></ul></div>
```

This behavior is caused by code in source/NVDAObjects/IAccessible/ia2TextMozilla.py, line 97. It is commented:
```
# The caret is at the end of an inline object.
# This will report "blank", but we want to report the character just after the caret.
```
And then we re-fetch the caret textInfo using _findNextContent so that it is truly the character after the caret.
However, this means in the case of the caret being at the end of an inline link at the end of a line, the character is reported as the first character of the next line, and expanding to line expands to the next line, not the current.

### Description of how this pull request fixes the issue:
A new limitToInline boolean keyword argument has been added to MozillaCompoundTextInfo._findNextContent, which when true, causes the method not to go above the deepest non-inline element when searching for the next object.
So in the case of a link at the end of a list item, it will not search outside of the list item.
This allows the work-around to still work at the end of links in the middle of a list item or paragraph, but no longer at the end, where it was inappropriate to do so.

### Testing performed:
* Created a list in Google Docs that contained a link at the end of the top item. Moved the caret to the end of the link by pressing end, and confirming the current line (top list item) was announced when reading the current line with NVDA. Before this change, the next line would be announced. Pressing end now also says "blank" rather than the first character of the next line.
* Opened the above html fragment in Firefox, switched to focus mode, and performed the same test as in Google Docs.

I can produce a system test once prs #11605  and #11569 are merged as these change the output of lists in contenteditables.

### Known issues with pull request:
None.

### Change log entry:
Bug fixes:
- NVDA now reads the correct line when the caret is placed at the end of a link on the end of a list item in Google Docs.

### CODE REVIEW

**Review by feerrenrut — DISMISSED** [review]
As suggested in the description for this PR, I think it would be good to hold off on merging and also include a system test.

**Comment by michaelDCurran** [comment]
@feerrenrut  I've added a system test to this now for you to review please. I confirmed that the system test breaks if I revert the actual change.

File: tests/system/robot/chromeTests.py [hunk 1]
```diff
@@ -170,3 +170,55 @@ def test_i7562():
...
+	)
+	# Move to the end of the line (which is also the end of the second link)
+	# Before pr #11606 this would have announced the bullet on the next line.
+	actualSpeech = _chrome.getSpeechAfterKey("end")
+	_asserts.strings_match(
+		actualSpeech,
+		"link"
+	)
+	# Read the current line.
+	# Before pr #11606 the next line would have been read.
```
Comment (feerrenrut) [inline tests/system/robot/chromeTests.py:219]: Just for clarity you could mention the contents of the next line:
```suggestion
	# Before pr #11606 the next line ("C D")  would have been read.

```

**Review by feerrenrut — DISMISSED** [review]
Looks good, a minor comment clarification that you could make, but happy for you to go ahead.

**Pull request merged**